### DISCUSSION TITLE
Ability to re-pitch a recording to a MIDI or hertz value

### DISCUSSION THREAD

**Discussion by RustoMCSpit** [godotengine/godot-proposals#8230] [Engine Core]
### Describe the project you are working on

https://github.com/dulvui/pocket-broomball/issues/11

Trying to implement a music mode which randomly picks a MIDI / hertz value to repitch a sound to a random note in the C major scale

### Describe the problem or limitation you are having in your project

See link.

Not a feature

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Introducing this would enable many music projects as these are the standard

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

midiCmaj = (c major notes in list)

Pitch meow.wav to midiCmaj[rand(0-6]

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

No, this is a vital piece for music

**Comment by Calinou** [comment]
Have you looked at the [MIDI Piano](https://github.com/godotengine/godot-demo-projects/tree/master/audio/midi_piano) demo project? It can play notes for the entire piano from a single audio sample file using pitch shifting.

**Reply by RustoMCSpit** [reply]
does it include microtonal notes? also seems like a bloated 'solution'(?).  at first glance it just looks like an audio sampler with 88 note pitch shifting which i guess works but would this not make more sense as a mutable property of a node?

**Reply by RustoMCSpit** [reply]
also, this only covers MIDI - not hertz
